### PR TITLE
Add how-to for using sample genotypes as covariates

### PR DESCRIPTION
**Pull request by tpoterba**

### CODE REVIEW

File: hail/python/hail/docs/guides/genetics.rst [hunk 1]
```diff
@@ -315,6 +315,34 @@ Multiple Phenotypes
...
+
+    Convert a Python list of variant strings to a Hail set literal:
+
+    >>> my_snps = ['1:1:A:T', '16:29102912:T:TTT']
+    >>> mt_snps_lit = hl.literal({hl.parse_variant(x) for x in my_snps})
+
+    Compute an array of genotypes per sample and annotate the matrix table `mt`:
+
+    >>> mt_filt = mt.filter_rows(mt_snps_lit.contains(mt.row_key))
+    >>> sample_genos = mt_filt.annotate_cols(genotypes = hl.agg.collect(mt_filt.GT.n_alt_alleles()))
```
Comment (jigold) [inline hail/python/hail/docs/guides/genetics.rst]: What is the type of `genotypes`? How are both variants accounted for?

File: hail/python/hail/docs/guides/genetics.rst [hunk 2]
```diff
@@ -315,6 +315,34 @@ Multiple Phenotypes
...
+    >>> mt = mt.annotate_cols(snp_covs = sample_genos[mt.x].genotypes)
```
Comment (jigold) [inline hail/python/hail/docs/guides/genetics.rst]: Where did `mt.x` come from?

File: hail/python/hail/docs/guides/genetics.rst (hunk 1 above)
Comment (tpoterba) [reply hail/python/hail/docs/guides/genetics.rst]: this is an array

File: hail/python/hail/docs/guides/genetics.rst (hunk 2 above)
Comment (tpoterba) [reply hail/python/hail/docs/guides/genetics.rst]: oops, meant to be mt.s

**Comment by jigold** [comment]
You still have an error in there.

**Comment by jigold** [comment]
@tpoterba Can we get this in this week or close this?

**Comment by tpoterba** [comment]
yes, it's quite annoying to deal with the doctests

**Comment by tpoterba** [comment]
Should be fixed now. I added a function `variant_str` which I use to improve the algorithm

File: hail/python/hail/expr/functions.py [hunk 3]
```diff
@@ -829,6 +830,62 @@ def parse_variant(s, reference_genome: Union[str, ReferenceGenome] = 'default')
     return _func('LocusAlleles({})'.format(reference_genome.name), t, s)
 
 
+def variant_str(*args) -> 'StringExpression':
+    """Create a variant colon-delimited string.
+
+    Parameters
+    ----------
+    args
```
Comment (jigold) [inline hail/python/hail/expr/functions.py]: Can this be more descriptive or delete?

**Comment by jigold** [comment]
I wanted to check the docs output, but I'm never going to catch this when it's not pending...

**Comment by jigold** [comment]
For future reference, the docs are here:
https://storage.googleapis.com/hail-ci-0-1/ci/3fa29bfca7bd2a3775533636b77e5c5de507d09c/c1d08a7fd26dab53bda8a682564463722812ab70/index.html

@tpoterba You're missing adding `variant_str` to the functions index page.

**Pull request merged**